File: backend/src/main/java/com/d4k/ecommerce/modules/order/service/OrderService.java
package com.d4k.ecommerce.modules.order.service;

import com.d4k.ecommerce.modules.order.dto.request.CancelOrderRequest;
import com.d4k.ecommerce.modules.order.dto.request.CreateOrderRequest;
import com.d4k.ecommerce.modules.order.dto.request.UpdateOrderStatusRequest;
import com.d4k.ecommerce.modules.order.dto.response.OrderResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Order Service Interface
 * Định nghĩa business logic cho orders
 */
public interface OrderService {
    
    /**
     * Tạo order từ cart của user
     * @param userId user ID
     * @param request thông tin order
     * @return order đã tạo
     */
    OrderResponse createOrder(Long userId, CreateOrderRequest request);
    
    /**
     * Lấy danh sách orders của user
     * @param userId user ID
     * @param pageable thông tin phân trang
     * @return danh sách orders
     */
    Page<OrderResponse> getUserOrders(Long userId, Pageable pageable);
    
    /**
     * Lấy chi tiết order
     * @param orderId order ID
     * @param userId user ID (để check ownership)
     * @return order detail
     */
    OrderResponse getOrderById(Long orderId, Long userId);
    
    /**
     * Hủy order (User)
     * @param orderId order ID
     * @param userId user ID (để check ownership)
     * @param request thông tin hủy
     */
    void cancelOrder(Long orderId, Long userId, CancelOrderRequest request);
    
    /**
     * Lấy tất cả orders (Admin)
     * @param pageable thông tin phân trang
     * @return danh sách orders
     */
    Page<OrderResponse> getAllOrders(Pageable pageable);
    
    /**
     * Lấy chi tiết order (Admin)
     * @param orderId order ID
     * @return order detail
     */
    OrderResponse getOrderByIdAdmin(Long orderId);
    
    /**
     * Update order status (Admin)
     * @param orderId order ID
     * @param request thông tin update
     * @return updated order
     */
    OrderResponse updateOrderStatus(Long orderId, UpdateOrderStatusRequest request);
    
    /**
     * Search orders (Admin)
     * @param keyword từ khóa tìm kiếm
     * @param pageable thông tin phân trang
     * @return danh sách orders
     */
    Page<OrderResponse> searchOrders(String keyword, Pageable pageable);
    /**
     * Update order status after payment (System)
     * @param orderId order ID
     * @param success payment success status
     */
    void updateOrderAfterPayment(Long orderId, boolean success);
}

File: frontend/src/services/order-service.js
import apiClient from './api-client';

const orderService = {
  getMyOrders: async (params) => {
    const response = await apiClient.get('/orders', { params });
    return response;
  },

  getOrderById: async (id) => {
    const response = await apiClient.get(`/orders/${id}`);
    return response;
  },

  createOrder: async (data) => {
    const response = await apiClient.post('/orders', data);
    return response;
  },

  cancelOrder: async (id, reason) => {
    const response = await apiClient.put(`/orders/${id}/cancel`, { cancelReason: reason });
    return response;
  },
  
  // ============ ADMIN METHODS ============

  getAllOrders: async (params = {}) => {
    const response = await apiClient.get('/admin/orders', { params });
    return response;
  },

  searchOrders: async (keyword, params = {}) => {
    const response = await apiClient.get('/admin/orders/search', {
      params: { keyword, ...params }
    });
    return response;
  },

  getOrderByIdAdmin: async (id) => {
    const response = await apiClient.get(`/admin/orders/${id}`);
    return response;
  },

  updateOrderStatus: async (id, status) => {
    const response = await apiClient.put(`/admin/orders/${id}/status`, { status });
    return response;
  }
};

export default orderService;
